src/metabase/util/grouper.clj

```diff
@@ -28,5 +43,9 @@
   "A wrapper of [[grouper.core/submit!]] that returns nil instead of a promise.
   We use grouper for fire-and-forget scenarios, so we don't care about the result."
   [& args]
-  (apply grouper/submit! args)
-  nil)
+  (let [p (apply grouper/submit! args)]
+    (when disable-grouper-batch-processing
+      ;; wake up the group immediately and wait for it to finish
+      (.wakeUp (first args))
+      (deref p))
```

Comment:
Putting aside whether we want to allow this behavior, I think it's surprising that "disable batching" just causes us to batch synchronously. It's possible for race to result in the batch still containing.

I also feel a bit wary of naming the setting after the underlying library, we might change that implementation detail.

WDYT about renaming the setting to "synchronous-batch-updates"? I don't think we have any other batch tasks, so we could also let the caller to this namespace decide, with a method like `process!` and call it something like "synchronous-metadata-updates".

Reply:
yea it's not ideal to do it like this. Another approach is to call the `body-fn` that was given to `grouper` when we `start!` one.

But that'd require us to modify the library.

I could do that as a follow-up since this is for testing purposes only.

Reply:
What do you think about the rename or refactoring I suggest? I don't think we should change the library.

Reply:
I agree with the rename but I don't fully understand your refactoring idea, could you be more specific?

Reply:
Let's just stick with the rename then and get it merged 😄 

I realize I didn't actually outline a non-invasive refactor, oops so much for remembering what I type on a weekend.

The idea would be to take all the places that call submit and rather call a new `process-X!` method defined where you build the queues. These methods would then use the setting to decide whether to call `submit!` or to just call the method that we register with the queue directly.

Reply:
> The idea would be to take all the places that call submit and rather call a new process-X! method defined where you build the queues. These methods would then use the setting to decide whether to call submit! or to just call the method that we register with the queue directly.

this makes sense but we currently don't have any way to get the processing method from the `Grouper` instance. So that's why I think we need to modify the lib to achieve that.

Reply:
renamed in [339f4e3](https://github.com/metabase/metabase/pull/45007/commits/339f4e342e148381d7c2d7500a3cb676b4e93b62)

Reply:
I think where we're missing each other is that I would put the branch up in the caller, not in `submit!`. I don't think introspecting a queue for its consumers is  good design.

```clojure
(defn- increment-view-counts!
  "Increment the view count of the given `model` and `model-id`."
  [model model-id]
  (if (synchronous-batch-updates)
    (increment-view-counts!* [{:model model :id :model-id}])
    (grouper/submit! @increase-view-count-queue {:model model :id model-id})))
```

Reply:
Let's leave that out of scope though, resolving this thanks to the rename.